rfcs/persistent_conversations_q_cli

```diff
@@ -0,0 +1,90 @@
+- Feature Name: persistent_conversations_q_cli
+- Start Date: 2024-03-27
+
+# Summary
+
+Enhance Amazon Q CLI Chat with persistent conversation management capabilities, enabling users to save, load, and manage conversations across sessions. This feature will initially support local storage with automatic conversation continuation based on the working directory, with future extensibility for cloud storage options.
+
+# Motivation
+
+The current ephemeral nature of Amazon Q CLI conversations presents several challenges. When a terminal closes, all conversation history is lost, forcing users to start from scratch. This lack of continuity is particularly problematic for people intermittently working on projects, where the context from previous hours or days is crucial. Moreover, it hampers the sharing of complex troubleshooting or design sessions with colleagues, especially across different time zones. The current setup also limits the potential for automation through scripting and impedes efficient knowledge transfer among team members. By introducing conversation persistence, we can address these issues, providing continuity, enabling collaboration, and supporting automation scenarios.
+
+
+The addition of conversation persistence will address these issues by providing continuity, enabling collaboration, and supporting automation scenarios.
+
+# Guide-level explanation
+
+The persistent conversations feature introduces several new capabilities to Amazon Q CLI:
+
+1. Automatic Continuation:
+```bash
+$ q
+Welcome to Amazon Q CLI!
+
+Would like to continue your previous conversation? (y/n)
+```
+    
+
+    
+2. Named Conversations:
+    
+```
+# Create/load a named conversation
+$ q --conversation my_project
+
+# List available conversations
+$ q --list-conversations
+Available conversations:
+1. my_project (Last modified: 2024-12-15 14:30)
+2. debug_lambda (Last modified: 2024-12-14 09:45)
+
+# Delete a conversation
+$ q --delete-conversation debug_lambda
+
+```
+
+    
+3. In-Chat Commands:
+* /conversation <name>: Create or load a conversation
+* /list-conversations: Show available conversations
+* /delete-conversation <name>: Remove a saved conversation
+
+# Reference-level explanation
+
+## Storage Implementation:
+
+We will use SQLite for conversation storage, located in the ~/.aws/amazonq/conversations/ directory. Each conversation will be stored in a file named using the SHA-1 hash of the working directory for automatic continuation. The schema will include the conversation history (with line limits), profile reference, context reference, and metadata such as timestamps.
+
+## Conversation Management:
```

Comment:
How do conversations interact with projects?

As a user, 90% of what I personally want here is just a single ongoing conversation and set of context files per dir. The only commands I want to use most of the time would be:

```
q  # the dir determines the project and what context I'm using
/clear  # explicitly start a new conversation. The default is to preserve the conversation
/history # See this entire conversation
```

It would be nice to explicitly select projects and spin up multiple projects/conversations in same directory on occasion, but it's the minority case.


(n fact, I did build it more or less this way for https://github.com/ANorwell/claippy)